scripts/addr2line.py

```diff
@@ -83,6 +87,87 @@ def __call__(self, address):
         self._input.stdin.flush()
         return self._read_resolved_address()
 
+class KernelResolver:
+    """A resolver for kernel addresses which tries to read from /proc/kallsyms."""
+
+    LAST_SYMBOL_MAX_SIZE = 1024
+
+    def __init__(self):
+        syms : list[tuple[int, str]] = []
+        ksym_re = re.compile(r'(?P<addr>[0-9a-f]+) (?P<type>.+) (?P<name>\S+)')
+        warnings_left = 10
+
+        self.error = None
+
+        try:
+            f = open('/proc/kallsyms', 'r')
```

Comment:
Note: would be nice to make this a parameter, so we can decode remotely (assuming we captured kallsyms). Not a blocker for the series.